x-pack/plugins/maps/public/connected_components/toolbar_overlay/tools_control/tools_control.tsx

```diff
@@ -76,38 +78,44 @@ export class ToolsControl extends Component<Props, State> {
 
   _closePopover = () => {
     this.setState({ isPopoverOpen: false });
+    if (this.props.filterModeActive) {
+      this.props.deactivateDrawMode();
```

Comment:
Should this just call this.props.cancelDraw instead of requiring a new prop "deactivateDrawMode". How about getting ride of deactivateDrawMode from props.